### PR TITLE
test: add hello-world-tab docker sample e2e test

### PR DESCRIPTION
**Pull request by Yimin-Jin**
Here is the successful workflow:
https://github.com/OfficeDev/TeamsFx/actions/runs/8829835162

### CODE REVIEW

File: packages/tests/src/commonlib/containeraAppValidator.ts [hunk 1]
```diff
@@ -46,27 +59,36 @@ export class ContainerAppValidator {
...
+      chai.assert.exists(
+        response,
+        `Response for ${containerAppName} should exist`
+      );
+    }
+
     console.log("Successfully validate Azure Container App Provision.");
   }
 
   static async validateContainerAppStatus(): Promise<void> {
```
Comment (hund030) [inline packages/tests/src/commonlib/containeraAppValidator.ts:78]: Why is this method static?

File: packages/tests/src/commonlib/containeraAppValidator.ts [hunk 2]
```diff
@@ -32,9 +34,20 @@ export class ContainerAppValidator {
...
+      EnvConstants.BACKEND_APP_NAME,
+      EnvConstants.FRONTEND_APP_NAME,
+    ].filter((name) => this.ctx[name] && this.ctx[name].length > 0)
+    .map((name) => this.ctx[name]);
+
+    chai.assert.isTrue(
+      this.containerAppNames && this.containerAppNames.length > 0
+    );
+
+    process.env[ContainerAppValidator.CONTAINER_NAME_LIST] = JSON.stringify(
```
Comment (hund030) [inline packages/tests/src/commonlib/containeraAppValidator.ts:48]: Avoid Updating global environment variables, which may affect other test cases.

File: packages/tests/src/commonlib/utilities.ts [hunk 3]
```diff
@@ -67,6 +67,19 @@ export function getKeyVaultNameFromResourceId(
   return result;
 }
 
+export function getContainerNameFromResourceId(resourceId: string): string {
```
Comment (hund030) [inline packages/tests/src/commonlib/utilities.ts:70]: Is this function still necessary?

File: packages/tests/src/commonlib/containeraAppValidator.ts [hunk 4]
```diff
@@ -32,9 +34,20 @@ export class ContainerAppValidator {
...
     chai.assert.exists(this.rg);
-    this.containerAppName = this.ctx[EnvConstants.AZURE_CONTAINER_APP_NAME];
-    chai.assert.exists(this.containerAppName);
-    process.env[EnvConstants.AZURE_CONTAINER_APP_NAME] = this.containerAppName;
+    this.containerAppNames = [
+      EnvConstants.AZURE_CONTAINER_APP_NAME,
+      EnvConstants.BACKEND_APP_NAME,
+      EnvConstants.FRONTEND_APP_NAME,
+    ].filter((name) => this.ctx[name] && this.ctx[name].length > 0)
+    .map((name) => this.ctx[name]);
```
Comment (hund030) [inline packages/tests/src/commonlib/containeraAppValidator.ts:42]:
```
this.containerAppNames = [
      EnvConstants.AZURE_CONTAINER_APP_NAME,
      EnvConstants.BACKEND_APP_NAME,
      EnvConstants.FRONTEND_APP_NAME,
    ].map((name) => this.ctx[name])
    .filter(value => !!value);
```
May be better.

File: packages/tests/src/commonlib/containeraAppValidator.ts (hunk 2 above)
Comment (Yimin-Jin) [reply packages/tests/src/commonlib/containeraAppValidator.ts:48]: Ok, I'll figure out how to optimize this.

File: packages/tests/src/commonlib/utilities.ts (hunk 3 above)
Comment (Yimin-Jin) [reply packages/tests/src/commonlib/utilities.ts:70]: You are right, I will remove this code.

File: packages/tests/src/commonlib/containeraAppValidator.ts (hunk 4 above)
Comment (Yimin-Jin) [reply packages/tests/src/commonlib/containeraAppValidator.ts:42]: Good idea!

**Review by hund030 — APPROVED** [review]

**Review by LongOddCode — APPROVED** [review]

**Review by ayachensiyuan — APPROVED** [review]

**Pull request merged**